src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {forwardRef, Ref} from 'react';
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+import RBSheet from 'react-native-raw-bottom-sheet';
+import IconClose from '../../../assets/icon/IconClose';
+import {BottomSheetV2} from '../../../components/BottomSheet';
+import dimen from '../../../utils/dimen';
+import {fonts} from '../../../utils/fonts';
+import {COLORS} from '../../../utils/theme';
+import {Button} from '../../../components/Button';
+import useChatClientHook from '../../../utils/getstream/useChatClientHook';
+import UserItem from './UserItem';
+
+export type Follow = 'signed' | 'incognito' | '';
+
+interface BottomSheetFollowProps {
+  username: string;
+  profilePicture: string;
+  topicName: string;
+  isFollow: boolean;
+  setIsFollow: (isFollow: boolean) => void;
+  memberCount: number;
+  setMemberCount: (memberCount: number) => void;
+  onClose: () => void;
+  getTopicDetail: (topicName: string) => void;
+  followType: Follow;
+  setFollowType: (followType: Follow) => void;
+}
+
+const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RBSheet>) => {
+  const {
+    topicName,
+    memberCount,
+    setMemberCount,
+    isFollow,
+    setIsFollow,
+    followType,
+    setFollowType,
+    getTopicDetail,
+    onClose
+  } = props;
+  const {followTopic} = useChatClientHook();
+
+  const handleFollowTopic = async ({type: followTypeParam}: {type: Follow}) => {
+    const isIncognito = followTypeParam === 'incognito';
+
+    const followTypeCheck = (targetType: Follow) => followType === targetType || followType === '';
+
+    const follow = async (type: Follow) => {
+      setFollowType(type);
+      setIsFollow(true);
+      try {
+        await followTopic(topicName, isIncognito);
+        getTopicDetail(topicName);
+      } catch (error) {
+        if (__DEV__) {
+          console.log(error);
+        }
+      }
+    };
+
+    onClose();
+
+    if (isFollow) {
+      if (followTypeCheck('')) {
+        setMemberCount(memberCount - 1);
+      }
+    } else {
+      setMemberCount(memberCount + 1);
+    }
+
+    if (followTypeParam === 'signed' && followTypeCheck('incognito')) {
+      follow('signed');
+    } else if (followTypeParam === 'incognito' && followTypeCheck('signed')) {
+      follow('incognito');
+    }
+  };
+
+  const handleUnfollowTopic = async ({type}: {type: Follow}) => {
+    setFollowType('');
+    setIsFollow(false);
+    setMemberCount(memberCount - 1);
+    onClose();
+    try {
+      const isIncognito = type === 'incognito';
+      await followTopic(topicName, isIncognito);
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error);
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the `handleUnfollowTopic` function could also benefit from being broken down into smaller functions. This would improve readability and maintainability of your code.

```diff
-  const handleUnfollowTopic = async ({type}: {type: Follow}) => {
-    setFollowType('');
-    setIsFollow(false);
-    setMemberCount(memberCount - 1);
-    onClose();
-    try {
-      const isIncognito = type === 'incognito';
-      await followTopic(topicName, isIncognito);
-    } catch (error) {
-      if (__DEV__) {
-        console.log(error);
-      }
-    }
-  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->